File: medicnote-backend/src/main/java/com/medicnote/backend/service/DashboardService.java
package com.medicnote.backend.service;

import com.medicnote.backend.dto.dashboard.DoctorDashboardDTO;
import com.medicnote.backend.dto.dashboard.PatientDashboardDTO;

public interface DashboardService {

    DoctorDashboardDTO getDoctorDashboard(Long doctorId);

    PatientDashboardDTO getPatientDashboard(Long patientId);
}

File: medicnote-frontend/src/services/dashboardService.ts
import api from "./api";

export const dashboardService = {
    /**
     * 🔹 Doctor dashboard
     * Role: DOCTOR, ADMIN
     */
    getDoctorDashboard: () =>
        api.get("/api/dashboard/doctor/me"),

    /**
     * 🔹 Patient dashboard
     * Role: PATIENT, ADMIN
     */
    getPatientDashboard: () =>
        api.get("/api/dashboard/patient/me"),
};